Marlin/temperature.cpp

```diff
@@ -614,6 +668,12 @@ void manage_heater() {
       thermal_runaway_protection(&thermal_runaway_state_machine[e], &thermal_runaway_timer[e], current_temperature[e], target_temperature[e], e, THERMAL_PROTECTION_PERIOD, THERMAL_PROTECTION_HYSTERESIS);
     #endif
 
+    #if MAX_TEMP_OVERSTOOT_TIME > 0
+      for (int8_t e = -1; e < EXTRUDERS; e++) {
```

Comment:
Not sure why you would set your hot end to 80C, but I just tried it on my
machine with a JHead and it settled in a few seconds and only overshoots 2C.
```
Send: M104 S80.000000
Recv: ok
Send: M105
Recv: ok T:41.0 /80.0 B:25.0 /0.0 @:100%
Send: M105
Recv: ok T:68.3 /80.0 B:25.0 /0.0 @:100%
Send: M105
Recv: ok T:82.1 /80.0 B:25.1 /0.0 @:0%
Send: M105
Recv: ok T:81.2 /80.0 B:25.1 /0.0 @:0%
Send: M105
Recv: ok T:79.3 /80.0 B:25.1 /0.0 @:15%
Send: M105
Recv: ok T:79.6 /80.0 B:25.2 /0.0 @:13%
Send: M105
Recv: ok T:80.1 /80.0 B:25.2 /0.0 @:5%
Send: M105
Recv: ok T:80.0 /80.0 B:25.1 /0.0 @:9%
Send: M105
Recv: ok T:80.0 /80.0 B:25.1 /0.0 @:11%
Send: M105
Recv: ok T:79.9 /80.0 B:25.1 /0.0 @:13%
Send: M105
Recv: ok T:80.2 /80.0 B:25.1 /0.0 @:3%
Send: M105
Recv: ok T:80.0 /80.0 B:25.0 /0.0 @:9%
Send: M105
Recv: ok T:80.1 /80.0 B:25.1 /0.0 @:6%
Send: M105
Recv: ok T:80.0 /80.0 B:25.1 /0.0 @:7%
Send: M105
Recv: ok T:80.0 /80.0 B:25.0 /0.0 @:8%
Send: M105
Recv: ok T:79.9 /80.0 B:25.1 /0.0 @:9%
Send: M105
Recv: ok T:80.1 /80.0 B:25.1 /0.0 @:5%
Send: M105
Recv: ok T:80.1 /80.0 B:25.2 /0.0 @:6%
Send: M105
Recv: ok T:80.0 /80.0 B:25.0 /0.0 @:6%
Send: M105
Recv: ok T:80.1 /80.0 B:25.1 /0.0 @:5%
Send: M105
Recv: ok T:80.2 /80.0 B:25.1 /0.0 @:2%
Send: M105
Recv: ok T:79.8 /80.0 B:25.1 /0.0 @:12%
```

On 17 May 2015 at 03:43, AnHardt <notifications@github.com> wrote:

> In Marlin/temperature.cpp
> <https://github.com/MarlinFirmware/Marlin/pull/2092#discussion_r30466844>:
>
> > @@ -614,6 +668,12 @@ void manage_heater() {
> >        thermal_runaway_protection(&thermal_runaway_state_machine[e], &thermal_runaway_timer[e], current_temperature[e], target_temperature[e], e, THERMAL_PROTECTION_PERIOD, THERMAL_PROTECTION_HYSTERESIS);
> >      #endif
> >
> > +    #if MAX_TEMP_OVERSTOOT_TIME > 0
> > +      for (int8_t e = -1; e < EXTRUDERS; e++) {
>
> @thinkyhead <https://github.com/thinkyhead>
> On the long term, beginning the arrays with the bed as 0 , and seeing as a
> heater as all the others, will save a lot of code in temperature.h.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/MarlinFirmware/Marlin/pull/2092/files#r30466844>.
>